### PR TITLE
[CARBONDATA-3149]Documentation for alter table column rename

### PR DESCRIPTION
**Pull request by akashrn5**
Add coumentation  for alter table column rename

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 NA
 - [x] Any backward compatibility impacted?
 NA
 - [x] Document update required?
YES
 - [x] Testing done
NA
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2108/

**Comment by SteNicholas** [comment]
@akashrn5 This document doesn't mention about comment of “alter table column rename”.Is document missing this point?And Spark 2.1 version couldn't support comment when alter table column rename.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2313/

**Comment by akashrn5** [comment]
@SteNicholas i did not get your point correctly, basicaly you are telling that,

1. column coment after column rename ?
2. whether spark 2.1 supports column rename?

am i right? if not, please can you explain it corectly, so that i can clear your doubt

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10362/

**Comment by SteNicholas** [comment]
@akashrn5 Yeah,for example
````
ALTER TABLE [db_name.]table_name CHANGE old_col_name new_col_name column_data_type comment ' '
````
Spark 2.1 couldn't support column coment after column rename.
[[BACKPORT] [SPARK-23525] [SQL] Support ALTER TABLE CHANGE COLUMN COMMENT for external hive table](https://github.com/apache/spark/pull/20768?from=singlemessage&isappinstalled=0)

File: docs/ddl-of-carbondata.md [hunk 1]
```diff
@@ -681,24 +682,28 @@ Users can specify which columns to include and exclude for local dictionary gene
...
+     - If the old column name and new column name are different and datatype is column's source datatype, it is just a column rename operation else it is both column rename and
+     datatype change operation.
 
      Valid Scenarios
      - Invalid scenario - Change of decimal precision from (10,2) to (10,5) is invalid as in this case only scale is increased but total number of digits remains the same.
      - Valid scenario - Change of decimal precision from (10,2) to (12,3) is valid as the total number of digits are increased by 2 but scale is increased only by 1 which will not lead to any data loss.
      - **NOTE:** The allowed range is 38,38 (precision, scale) and is a valid upper case scenario which is not resulting in data loss.
 
-     Example1:Changing data type of column a1 from INT to BIGINT.
+     Example1:Rename column a1 to a2 and changing data type from INT to BIGINT.
```
Comment (qiuchenjian) [inline docs/ddl-of-carbondata.md]:
```suggestion
     Example1:Rename column a1 to a2 and change data type from INT to BIGINT.
```

File: docs/ddl-of-carbondata.md [hunk 2]
```diff
@@ -681,24 +682,28 @@ Users can specify which columns to include and exclude for local dictionary gene
...
-     This command is used to change the data type from INT to BIGINT or decimal precision from lower to higher.
+     This command is used to change the data type from INT to BIGINT or decimal precision from lower to higher and rename column.
      Change of decimal data type from lower precision to higher precision will only be supported for cases where there is no data loss.
 
      ```
-     ALTER TABLE [db_name.]table_name CHANGE col_name col_name changed_column_type
+     ALTER TABLE [db_name.]table_name CHANGE old_col_name new_col_name column_data_type
      ```
+     Scenarios:
+     - If both the old and new column name are same, it is just a datatype change operation
```
Comment (qiuchenjian) [inline docs/ddl-of-carbondata.md]:
```suggestion
     - If both the old and new column names are the same, it is just a datatype change operation
```

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2109/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2314/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10363/

File: docs/ddl-of-carbondata.md [hunk 3]
```diff
@@ -681,24 +682,28 @@ Users can specify which columns to include and exclude for local dictionary gene
 
      **NOTE:** Drop Complex child column is not supported.
 
-   - ##### CHANGE DATA TYPE
+   - ##### RENAME COLUMN AND CHANGE DATATYPE
```
Comment (manishgupta88) [inline docs/ddl-of-carbondata.md]: Change this to `Change Column Name/Type`. Please go through the below link and try to align the documentation in that way if possible
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ChangeColumnName/Type/Position/Comment

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2154/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10408/

File: docs/ddl-of-carbondata.md [hunk 4]
```diff
@@ -47,7 +47,8 @@ CarbonData DDL statements are documented here,which includes:
     * [RENAME TABLE](#rename-table)
     * [ADD COLUMNS](#add-columns)
     * [DROP COLUMNS](#drop-columns)
-    * [CHANGE DATA TYPE](#change-data-type)
+    * [RENAME COLUMN](#change-column-name-/-type)
+    * [CHANGE DATA TYPE](#change-column-name-/-type)
```
Comment (manishgupta88) [inline docs/ddl-of-carbondata.md]: check for linking. With this text it will not link

File: docs/ddl-of-carbondata.md [hunk 5]
```diff
@@ -681,24 +682,24 @@ Users can specify which columns to include and exclude for local dictionary gene
 
      **NOTE:** Drop Complex child column is not supported.
 
-   - ##### CHANGE DATA TYPE
+   - ##### CHANGE COLUMN NAME/TYPE
    
-     This command is used to change the data type from INT to BIGINT or decimal precision from lower to higher.
+     This command is used to change the column's name and the data type from INT to BIGINT or decimal precision from lower to higher and rename column.
```
Comment (manishgupta88) [inline docs/ddl-of-carbondata.md]: Write this as below
`This command is used to change column name and the data type from INT to BIGINT or decimal precision from lower to higher.`

File: docs/ddl-of-carbondata.md [hunk 6]
```diff
@@ -681,24 +682,24 @@ Users can specify which columns to include and exclude for local dictionary gene
...
      Change of decimal data type from lower precision to higher precision will only be supported for cases where there is no data loss.
 
      ```
-     ALTER TABLE [db_name.]table_name CHANGE col_name col_name changed_column_type
+     ALTER TABLE [db_name.]table_name CHANGE old_col_name new_col_name column_data_type
```
Comment (manishgupta88) [inline docs/ddl-of-carbondata.md]: Change this as below
`ALTER TABLE [db_name.]table_name CHANGE col_old_name col_new_name column_type`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2161/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2374/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10415/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**